rfc/0010-rancher-integration-policies.md

```diff
@@ -0,0 +1,176 @@
+|              |                                  |
+| :----------- | :------------------------------- |
+| Feature Name | Rancher integration of Kubewarden Policies  |
+| Start Date   | 2022-08-22                       |
+| Category     | Development                      |
+| RFC PR       | https://github.com/kubewarden/rfc/pull/11  |
+| State        | **ACCEPTED**                     |
+
+
+# Summary
+[summary]: #summary
+
+This RFC explains how the Kubewarden policies, either owned by the Kubewarden
+team or 3rd party ones, can be integrated in Rancher Explorer.
+
+For policies owned by the Kubewarden team it proposes building Rancher Helm charts
+that will be published under https://github.com/rancher/charts.
+
+For 3rd party policies not owned by the Kubewarden team,
+it proposes using the Artifact Hub API to poll for the metadata stored in
+`policy/artifacthub-pkg.yml`.
+
+
+# Motivation
+[motivation]: #motivation
+
+Problems to solve:
+
+- Obtain the list of available policies regardless of where they come from, and
+  their metadata to show in Rancher Explorer UI.
+- Allow for groups of policies to be installed.
+- Implement it efficiently and with low development maintenance costs.
+- Allow for airgap installations.
+
+## Examples / User Stories
+[examples]: #examples
+
+1. As a user, I want to install one Kubewarden policy owned by the
+   Kubewarden/Rancher team via Rancher Explorer UI.
+2. As a Kubewarden developer, I want to allow users to deploy a collection of related policies
+   in a simple way. For example: "PSP best of", "Security best practices",...
+3. As a user, I want to install 3rd party Kubewarden policies via Rancher
+   Explorer UI.
+4. As a Kubewarden developer, I want to release a new Kubewarden chart/policy so
+   it can be installed via Rancher Explorer UI.
+5. As a 3rd party developer, I want to release a policy on Artifact Hub so it can
+   be discovered and installed via Rancher Explorer UI.
+
+# Detailed design
+[design]: #detailed-design
+
+## For policies owned by the Kubewarden team
+
+Follow the process listed in [RFC-9, Rancher integration of Kubewarden
+charts](./0009-rancher-integration-charts.md).
+
+Carry the Helm chart code for the policy in a `rancher-chart` folder in the repo.
+
+The resulting chart built from the policy repository will not be served and
+released by us, but used to build Rancher charts from source by using the
+[Package](https://github.com/rancher/charts/blob/dev-v2.6/docs/packages.md)
+format in https://github.com/rancher/charts.
+They will be submitted via a `packages/kubewarden/<policy>.package.yaml`, with
+`package.yaml::commit` pointing to the relevant `rancher-X` branch.
```

Comment:
Does it mean we will have our own fork of https://github.com/rancher/charts (maybe as a Kubewarden org)? I will assume we do.

In addition to the current tasks, before tagging a new policy release we will ensure the contents of the `rancher-chart` directory are uptodate. Then we will tag the new version.

After having done that, we will go to our fork of rancher charts and:
  * Create a branch for updating this policy
  * Copy the contents of `rancher-chart` from the policy into the right directory
  * Create a PR

Is that correct?

Reply:
Clarified over a call

Reply:
(as talked on call)

More or less. Submission into the Rancher charts repository happens by providing a [package.yaml](https://github.com/rancher/charts/blob/dev-v2.6/docs/packages.md), which accepts [several ways to point to an upstream chart](https://github.com/rancher/charts/blob/dev-v2.6/docs/packages.md#upstreamoptions). Rancher charts are built from the package definitions.

I would point to upstream policy charts using the github repo, subdirectory. We could have a bot that bumps those in a fork of https://github.com/rancher/charts.

Added explanation on https://github.com/kubewarden/rfc/pull/12/commits/fcf6b07eb44c2f0d9b9ce08fefb1f4e896b7bafd.